expression/scalar_function.go

```diff
@@ -258,25 +265,48 @@ func newFunctionImpl(ctx sessionctx.Context, fold int, funcName string, retType
 	return sf, nil
 }
 
+// ScalarFunctionCallBack is the definition of callback of calling a newFunction.
+type ScalarFunctionCallBack func(function *ScalarFunction) (Expression, error)
+
+func defaultScalarFunctionCheck(function *ScalarFunction) (Expression, error) {
+	// todo: more scalar function init actions can be added here, or setting up with customized init callback.
+	if function.FuncName.L == ast.Grouping {
+		if !function.Function.(*BuiltinGroupingImplSig).isMetaInited {
```

Comment:
What's metadata used for?

Reply:
grouping function's meta is a kind of BitMask used to compute the grouping value based on the gid column